mage/query-modules/python/elasticsearch-synchronization.md

```diff
@@ -0,0 +1,288 @@
+- `max_backoff: float` -> The maximum number of seconds a retry will wait.
+- `yield_ok: float` -> If set to `False` will skip successful documents in the output.
+- `thread_count: int` -> **Size of the threadpool** to use for the bulk requests.
+- `queue_size: int` -> Size of the **task queue** between the **main thread (producing chunks to send) and the processing threads**.
+
+The method could be called in a following way:
+```
+CALL elastic_search_serialization.index_db("node_index", "edge_index", 5, 256, 104857600, True, False, 2, 2.0, 600.0, True, 2) YIELD *;
+```
+
+#### **Output**
+- `number_of_nodes: int` -> Number of indexed nodes.
+- `number_of_edges: int` -> Number of indexed edges.
+
+### `reindex()`
+**Reindex all documents** that satisfy a given query from one index to another, potentially (if target_client is specified) on a different cluster. If you don’t specify the query you will reindex all the documents.
+
+#### **Input**
+- `updatatedObjects: List[Dict[str, Any]]` -> List of all objects that **were updated and then sent as arguments to this method** with the help of the "update trigger".
+- `source_index: Union[str, List[str]])` -> Identifies **source index(or more of them)** from where documents need to be indexed.
+- `target_index: str` -> Identifies **target index** to where documents need to be indexed.
+- `query: str` -> Query written as JSON.
+- `chunk_size: int` -> Number of docs in one chunk sent to es (default: 500).
+- `scroll: str` -> Specifies how long **a consistent view of the index** should be maintained for scrolled search.
+- `op_type: Optional[str])` -> Explicit operation type. Defaults to ‘_index’. Data streams must be set to ‘create’. If not specified, will auto-detect if target_index is a data stream.
+
+
+#### **Output**
+- `response: str` -> Number of documents matched by a query in the source_index.
+
+To reindex all documents from the `source_index` to the `destination_index`, use the following query:
+```
+CALL elastic_search_serialization.reindex("source_index", "destination_index", "{\"query\": {\"match_all\": {}}} ") YIELD * RETURN *;
+```
+
+### `scan()`
+Fetches documents from the index specified by the index_name matched by the query. Supports pagination.
+
+#### **Input**
+- `index_name: str` -> Name of the index.
+- `query: str` -> Query written as JSON.
+- `scroll: int` -> Specifies how long **a consistent view** of the index should be maintained for scrolled search.
+- `raise_on_error: bool` -> Raises an exception (`ScanError`) if an error is encountered (some shards fail to execute). By default, it's raised.
+- `preserve_order: bool` -> Don’t set the search_type to scan - this will cause the scroll to paginate with preserving the order. Note that this can be an extremely expensive operation and can easily lead to unpredictable results, use it with caution.
```

Comment:
Where do I set this search_type? 

Maybe first explain what this does, then give the recommendations and notes.

-> Preserves the order of SOMETHING. Don's use with search_type set to scan because the scroll will paginate with preserving the order. 

Something like that, i have no idea what any of this means but 1. give a description of the argument 2. clarify the search_type (i can't find it anywhere in this page except here)